kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/SiteEmailNotificationContent.java

```diff
@@ -80,13 +80,15 @@
     protected static FormattedText getFormattedText() {
         if (formattedText == null) {
             synchronized (LOCK) {
-                FormattedText component = (FormattedText) ComponentManager.get(FormattedText.class);
-                if (component == null) {
-                    throw new IllegalStateException("Unable to find the FormattedText using the ComponentManager");
-                } else {
-                    formattedText = component;
-                }
-            }
+		if(formattedText == null){
```

Comment:
This is the standard solution to the problem, there's more about the double lock problem here: http://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html

However the effect of getting multiple copies of the service from the ComponentManager shouldn't matter as the same object will get returned each time, so although the code isn't perfect as it was the impact of it getting multiple copies is debatable.